test/integration/backward_compatible/parallel/HazelcastClientTest.js

```diff
@@ -71,28 +70,37 @@ const configParams = [
     smartConfig
 ];
 
+const testFactory = new TestUtil.TestFactory();
+
 configParams.forEach((cfg) => {
     describe('HazelcastClientTest[smart=' + cfg.network.smartRouting + ']', function () {
         let cluster;
         let client;
         let managed;
+        let member;
 
         const filterInternalMaps = (distributedObjects) => {
             return distributedObjects.filter(distObj => !distObj.getName().startsWith('__'));
         };
 
+        before(async function () {
+            cluster = await testFactory.createClusterForParallelTests();
+            member = await RC.startMember(cluster.id);
+        });
+
         beforeEach(async function () {
-            cluster = await RC.createCluster(null, null);
-            await RC.startMember(cluster.id);
-            cfg.clusterName = cluster.id;
-            client = await Client.newHazelcastClient(cfg);
             managed = new ManagedObjects();
+            cfg.clusterName = cluster.id;
+            client = await testFactory.newHazelcastClientForParallelTests(cfg, member);
         });
 
         afterEach(async function () {
             await managed.destroyAll();
             await client.shutdown();
-            await RC.terminateCluster(cluster.id);
+        });
+
+        after(async function () {
+            await testFactory.shutdownAll();
```

Comment:
this is needed since there is a test that shutdown the client `more than one call to shutdown returns same promise`